Add filename to saveFilePrompt.

BugsNash opened the pull request:
Note that this is a draft patch, I would like a first pass review please.

I am blocked on getting translated messages for the messages files, which is required before this patch can be merged.

I'm open to changing the design for the maximum length of file name. Currently it looks like this for long file names when the window is resized to the smallest possible size:
![screenshot from 2018-04-24 14-47-44](https://user-images.githubusercontent.com/6046079/39166775-d8e0339a-47ce-11e8-9d87-20e1f6490923.png)

beaufortfrancois commented on _locales/en/messages.json:
```diff
@@ -158,8 +158,8 @@
   "saveFilePrompt": {
-    "message": "Do you want to save the file before closing?",
-    "description":"The text which prompts user if they want to save file before closing it."
+    "message": " has been modified. Do you want to save it before closing?",
```
You may want to use placeholder instead of concatenating strings. See https://developer.chrome.com/apps/i18n

BugsNash replied:
that's way better, thanks :)

BugsNash commented:
This is ready for final review now :)

BugsNash commented:
Oh and the translations for the non English messages were sourced through Google's internal localization team.

BugsNash commented:
I have changed the ellipsis implementation from a hard coded javascript limit to css text-overflow, which allows more of the filename to be visible when the window is larger. This wasn't possible with -webkit-box on the dialog-text element or it's parent, so I removed it (and associated -webkit-box properties) from them both. They didn't seem to be necessary and haven't changed the rendering at all. 

Updated screenshots:

![screenshot from 2018-05-17 13-38-51](https://user-images.githubusercontent.com/6046079/40155718-9efab742-59d8-11e8-9c9a-f3d96ab6baf9.png)
![screenshot from 2018-05-17 13-39-10](https://user-images.githubusercontent.com/6046079/40155720-9f27e06e-59d8-11e8-9656-535ff3e6b4a3.png)

BugsNash commented on js/tabs.js:
```diff
@@ -262,10 +262,11 @@ Tabs.prototype.close = function(tabId) {
   }
 
   var tab = this.tabs_[i];
+  var filename = tab.getEntry().name;
```
Thank you! I had actually fixed this already but it got lost in a rebase conflict :\ Would not have noticed

BugsNash commented on _locales/en/messages.json:
```diff
@@ -158,8 +158,14 @@
   "saveFilePrompt": {
-    "message": "Do you want to save the file before closing?",
-    "description":"The text which prompts user if they want to save file before closing it."
+    "message": "$filename$ has been modified. Do you want to save it before closing?",
```
lgtm :)

BugsNash replied:
Is it ok if I do this in a follow up patch? This will probably require waiting on another round of translation as I will need to split the prompt in the messages file into 2 and get a new description translated

beaufortfrancois replied:
Sure.
Please file an issue so that we can keep track of it.

BugsNash commented on _locales/en/messages.json:
```diff
@@ -158,8 +158,14 @@
+    "placeholders": {
+      "filename": {
+        "content": "$1",
+        "example": "index.html"
```
Ack

BugsNash replied:
Actually if I change it to a meaningfully named .txt file (like myfile.txt) it would require another round of translation. I figured html files would be a common use case for the app (and the name of the index.html name wouldn't change in different languages). Maybe I'll just use something like 'file.txt' and hope that Google translate is good enough since this is example text only that the user doesn't see.

beaufortfrancois commented:
LGTM
